### PR TITLE
Make sure `got.mergeOptions(...)` can be assigned to defaults

### LINKED ISSUE #993: Cannot assign `got.mergeOptions(...)` to instance defaults
#### What would you like to discuss?

Even if I set `mutableDefaults` as true, (TS 3.7.3, got 10.0.4), TS still complains that `defaults` is a read-only property. The readme example of the `afterResponse` hook doesn't seem to work when using TS.

According to the definition file, `defaults` may or may not be `ReadOnly`.

If I just invoke `client.mergeOptions(client.defaults.options, updatedOptions);` instead of reassigning `client.defaults.options` it seems to work fine but in that case it works even if I don't set `mutableDefaults` as true.

Any ideas or have I just misunderstood how it works?

#### Checklist

- [ x] I have read the documentation.

### PR DESCRIPTION
**Pull request by szmarczak**
Fixes #993

### CODE REVIEW

File: source/types.ts [hunk 1]
```diff
@@ -202,10 +203,44 @@ export interface NormalizedOptions extends Options {
...
+	followRedirect: boolean;
+	useElectronNet: boolean;
+	methodRewriting: boolean;
+	context: {[key: string]: any};
+
 	// UNIX socket support
 	path?: string;
 }
 
+/*
```
Comment (szmarczak) [inline source/types.ts]: Cannot decide which is better. Both solutions work. Should be `NormalizedOptions` inherited from `Options` or `DefaultOptions`?

Comment (sindresorhus) [reply source/types.ts]: Semantically, it makes more sense for it to inherit from `Options`, but I might be missing some context. Are there any downsides/upsides to either way?

Comment (szmarczak) [reply source/types.ts]: It's just that `DefaultOptions` are already normalized but some properties are **excluded** there. 

I agree. Let's go with `Options`.

File: source/create.ts [hunk 2]
```diff
@@ -56,7 +56,7 @@ interface GotFunctions {
 
 export interface Got extends Record<HTTPAlias, GotFunctions>, GotFunctions {
 	stream: GotStream;
-	defaults: Defaults | Readonly<Defaults>;
+	defaults: Defaults;
```
Comment (szmarczak) [inline source/create.ts]: No point in having `Defaults | Readonly<Defaults>` as in some cases TS misguesses it as `Readonly<Defaults>` and people have to cast it to `Defaults` anyway. The JS runtime will throw anyway if anyone tries to modify frozen defaults.

**Pull request merged**